dbt/include/spark/macros/materializations/incremental/strategies.sql

```diff
@@ -2,8 +2,14 @@
 
     {%- set dest_columns = adapter.get_columns_in_relation(target_relation) -%}
     {%- set dest_cols_csv = dest_columns | map(attribute='quoted') | join(', ') -%}
-    insert overwrite table {{ target_relation }}
-    {{ partition_cols(label="partition") }}
+    {% if target_relation.is_iceberg %}
+      {# removed table from statement for iceberg #}
```

Comment:
If I’m not mistaken, the primary difference here is that Iceberg supports the catalog v2 api and hudi and delta are using the data catalog v1 api. If that’s the case, maybe we should emphasize that as opposed to just the format types.

That’s the difference that needs to be looked up or updated in Spark to support use cases etc.

Reply:
Actually looking at the rest, it seems like just emphasizing the specifics for now would be more consistent in this PR.

Anything not user facing can be updated (unless that’s frowned upon here on internal code too much). I haven’t checked how package private any of this might be.

But we might choose to also document it now if you’d like. Or it can be done as a follow up. Keeping the format specific things is much more in line with how the library currently works it seems.